camel/memories/blocks/graphdb_block.py

```diff
@@ -0,0 +1,224 @@
+                        value = {}  # Convert back to an empty dictionary
+                    if key == 'role_at_backend':
+                        value = OpenAIBackendRole[value]
+
+                    # Handle only message-related keys
+                    if key.startswith('message_'):
+                        sub_key = key[len('message_') :]
+                        if sub_key == 'role_type':
+                            value = RoleType[
+                                value
+                            ]  # Convert back to Enum if applicable
+                        message[sub_key] = value
+                    else:
+                        reconstructed[key] = value
+
+                reconstructed['message'] = message
+                return reconstructed
+
+            return [
+                ContextRecord(
+                    memory_record=MemoryRecord.from_dict(
+                        reconstruct_from_flat(result["n"])
+                    ),
+                    score=result.get("similarity", 0),
+                )
+                for result in results
+            ]
+        except Exception as e:
+            logger.error(f"Error retrieving records: {e}")
+            return []
+
+    def write_records(self, records: List[MemoryRecord]) -> None:
+        """Writes recordes to the graph database."""
+        for record in records:
+            properties = record.to_dict()
+
+            def flatten_and_convert(d):
+                for key, value in list(d.items()):
+                    if isinstance(value, Enum):
+                        d[key] = value.name  # Convert Enum to its name
+                    elif value is None:
+                        d[key] = (
+                            '__none__'  # Convert None to a sentinel string
+                        )
+                    elif isinstance(value, dict):
+                        if value:  # If dictionary is not empty
+                            for sub_key, sub_value in flatten_and_convert(
+                                value
+                            ).items():
+                                d[f"{key}_{sub_key}"] = sub_value
+                            del d[key]
+                        else:
+                            d[key] = '__empty_dict__'
+                    elif isinstance(value, list):
+                        d[key] = [
+                            str(i)
+                            if not isinstance(i, (str, int, float, bool))
+                            else i
+                            for i in value
+                        ]
+                    else:
+                        d[key] = (
+                            str(value)
+                            if not isinstance(value, (str, int, float, bool))
+                            else value
+                        )
+                return d
+
+            properties = flatten_and_convert(properties)
+
+            content = record.message.content
```

Comment:
here we add the message as one node to graph database, I feel it would be better if we can extract the entity from this message using `KnowledgeGraphAgent`, then add entity to the graph database. Since the relationship between entities in a message makes more value than the relationship between 2 messages. WDYT?